xcode/Ext-Safari/SafariWebExtensionHandler.swift

```diff
@@ -2,13 +2,30 @@ import SafariServices
 
 class SafariWebExtensionHandler: NSObject, NSExtensionRequestHandling {
 	func beginRequest(with context: NSExtensionContext) {
+		let lock = DispatchSemaphore(value: 0)
+
+		Task {
+			guard
+				let request = context.inputItems[0] as? NSExtensionItem,
```

Comment:
Yeah, I was looking for something like Rust's `block_on` but it seems like Swift really doesn't want you to do something like that, and it also seems like you can't control which thread an async task runs on. Hence this awful code. It should be safe though, considering the semaphore should ensure the object is only handled by one thread at a time.

Reply:
Right, this answer seems to me to specifically be about calling a blocking function inside an async context (from a thread used by the async runtime). Doing that can lead to a deadlock in probably any language with async functions, including Rust, if you have limited threads. I'm pretty sure beginRequest(with:) here is not called from an async context, though, so none of the async runtime threads will ever be blocked.

The semaphore is just in case the function calling beginRequest does something else with the NSExtensionContext afterwards. I don't assume so, it did work without it but dealing with data races in the past has made me careful :P